clients/cobol-lsp-vscode-extension/src/services/util/FSUtils.ts

```diff
@@ -84,26 +27,24 @@ export function getURIFromResource(resource: string): urlUtil.URL[] {
  * @param copybookFolders list of folders from where to search the copybook
  * @param extensions list of possible copybooks extensions
  */
-export function searchCopybookInWorkspace(
+export function searchCopybookInExtensionFolder(
   copybookName: string,
   copybookFolders: string[] | undefined,
   extensions: string[] | undefined,
 ): string | undefined {
-  for (const workspaceFolderPath of SettingsUtils.getWorkspaceFoldersPath()) {
-    if (!copybookFolders || !extensions) return undefined;
-    const workspaceFolder = cleanWorkspaceFolder(workspaceFolderPath);
-    for (const p of copybookFolders) {
-      for (const ext of extensions) {
-        const searchResult = globSearch(workspaceFolder, p, copybookName, ext);
-        if (searchResult) {
-          const root = path.parse(searchResult).root;
-          const urlPath = searchResult
-            .substring(root.length)
-            .split(path.sep)
-            .map((s) => encodeURIComponent(s))
-            .join(path.sep);
-          return new urlUtil.URL("file://" + root + urlPath).href;
-        }
+  if (!copybookFolders || !extensions) return undefined;
+  const extensionFolder = cleanWorkspaceFolderName(Utils.getExtensionFolder());
```

Comment:
I think that this change completely breaks the local copybook search (from settings list).

Reply:
It's a bad architecture but this is handled here : https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/pull/2265/files#diff-3382794271734bc35f6dd83c6852607f7cb2e1000cdb122b9a2a2bbb3b59489aL49 . 

And this also might leads an infinite loop if downloaded file can't be found.

Reply:
The place where @ishche asked `Are we backward compatible?` - the answer is `no` because of this change.

The `searchCopybookInExtensionFolder` probably needs to be parametrized (and renamed?).

Reply:
resolved at : a5652c20285e22779fc890ce4cb4275730e7e025